doc/toscalite/spec.md

```diff
@@ -0,0 +1,175 @@
+* ``Properties`` -  This block defines the node's properties. It has to contain property elements, each with an attribute Key. The attribute key is required to be a lower-case name without spaces (allowed characters `abcdefghijklmnopqrstuvxyz`).
+
+The ``Type`` and ``Name`` elements are always required. ``Properties`` are mandatory in machine nodes and optional in service nodes.
+
+### Machines
+
+A machine node represents a Linux-based system, usually Ubuntu 16.04. The deployment system does not check whether the system is running, as long as the machine is reachable with SSH. Therefore SSH has to be installed on every target machine.
+This specific type of node gets described as follows:
+
+```XML
+<Node>
+    <Type>machine</Type>
+    <Name>database</Name>
+    <Properties>
+        <Property key="ip">192.168.178.1</Property>
+        <Property key="username">root</Property>
+        <Property key="password">1234567</Property>
+    </Properties>
+</Node>
+```
+**Explanation:**
+* ``Type`` - For a machine node the type always has to be ``machine``. In a service node it would be ``service``.
+* `Name` - As specified earlier, a name refers to the unique name of a node. (See [Nodes](#nodes))
+* `Properties` - The properties block is mandatory in a machine node, because it is used to define the IP address (of the host), a username and password. The specific attribute keys for these mandatory properties can be taken from the example above.
+    * For simplicity TOSCAlite only supports SSH authentication based on username and password. Public/private key authentication is not supported.
+
+### Services
+
+A service represents the part of a topology that has to be installed on a machine or on top of other services. The steps to install, start or stop the service are defined in shell scripts (see [Implementation Artifacts](#implementation-artifacts)). Files other than shell scripts get defined in deployment artifacts.
+
+#### Implementation Artifacts
+
+Implementation artifacts have to implement the lifecycle operations for a service. Due to TOSCAlite requiring Linux-based systems, shell scripts have to be used to perform these operations.
+
+The following operations are supported:
+* Create - Mandatory for every service. Used to create the service. This script/executable should install all the packages required to run the service.
+* Start - Optional. Used to start the service.
+* Stop - Optional. Used to stop the service.
+
+##### Execution of Implementation Artifacts
+
+Before any script gets executed, the whole TOSCAlite archive is loaded onto every virtual machine and extracted there. All the extracted files will be located in the home directory of the user that is defined in the machine node.
+
+The scripts get executed in the directory they are located in. This means that the TOSCAlite deployment system will automatically change the parent working directory to the directory in which the script is located.
```

Comment:
> This means that the TOSCAlite deployment system **has to**  automatically change the parent working directory to the directory in which the script is located.

**Has to** because it is mandatory that the deployment system does that.